app/code/Magento/Cms/view/adminhtml/web/js/folder-tree.js

```diff
@@ -78,30 +79,92 @@ define([
             var path = this.options.currentPath,
                 tree = this.element,
                 lastExistentFolderEl,
+                pathId,
 
-                /**
-                 * Recursively open folders specified in path array.
-                 */
-                recursiveOpen = function () {
-                    var folderEl = $('[data-id="' + path.pop() + '"]');
+            /**
+             * Recursively open folders specified in path array.
+             */
+            recursiveOpen = function () {
+                var folderEl = $('[data-id="' + path.pop() + '"]');
 
-                    // if folder doesn't exist, select the last opened folder
-                    if (!folderEl.length) {
-                        tree.jstree('select_node', lastExistentFolderEl);
+                // if folder doesn't exist, select the last opened folder
+                if (!folderEl.length) {
+                    tree.jstree('select_node', lastExistentFolderEl);
 
-                        return;
-                    }
+                    return;
+                }
+
+                lastExistentFolderEl = folderEl;
+
+                if (path.length) {
+                    tree.jstree('open_node', folderEl, recursiveOpen);
+                } else {
+                    tree.jstree('open_node', folderEl, function () {
+                        tree.jstree('select_node', folderEl);
+                    });
+                }
+            },
+
+            /**
+             * Encode path value
+             *
+             * @param {String} val
+             */
+            _encodePathId = function (val) {
+                return Base64.encode(val)
+                    .replace(/\+/g, ':')
+                    .replace(/\//g, '_')
+                    .replace(/=/g, '-');
+            },
+
+            /**
+             * Decode path value
+             *
+             * @param {String} val
+             */
+            _decodePathId = function (val) {
+                return Base64.decode(val)
+                    .replace(/\:/g, '+')
+                    .replace(/\_/g, '/')
+                    .replace(/-/g, '=');
+            },
+
+            /**
+             * Get currentPath decode it returns new path array
+             */
+            _parseCurrentPath = function () {
+                var paths = [],
+                    decodedPath = _decodePathId(window.MediabrowserUtility.pathId.replace(/--|,,/, '==')).split('/');
 
-                    lastExistentFolderEl = folderEl;
+                $.each(decodedPath, function (i, val) {
+                    var isLastElement = i === decodedPath.length - 1;
 
-                    if (path.length) {
-                        tree.jstree('open_node', folderEl, recursiveOpen);
+                    if (isLastElement) {
+                        paths[i] = window.MediabrowserUtility.pathId.replace(',,', '--');
```

Comment:
By fixing the uploader button url we will be able to remove this replace as well.